### PR TITLE
tests: fs: check for arguments NULLness

### LINKED ISSUE #29704: [Coverity CID :215255] Dereference before null check in tests/subsys/fs/fs_api/src/test_fs.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/c9a2a5e7fb0194bfb168f5aa1a0a83c73f62acb3/tests/subsys/fs/fs_api/src/test_fs.c#L300

Category: Null pointer dereferences
Function: `temp_mount`
Component: Tests
CID: [215255](https://scan9.coverity.com/reports.htm#v29726/p12996/mergedDefectId=215255)

Details:

```
294     }
295    
296     static int temp_mount(struct fs_mount_t *mountp)
297     {
298         size_t len = strlen(mountp->mnt_point);
299    
>>>     CID 215255:  Null pointer dereferences  (REVERSE_INULL)
>>>     Null-checking "mountp" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.
300         if (mountp == NULL) {
301             return -EINVAL;
302         }
303    
304         if (mountp->mnt_point[len - 1] != ':') {
305             return -EINVAL;
```

Please fix or provide comments in coverity using the link:

https://scan9.coverity.com/reports.htm#v32951/p12996.

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### PR DESCRIPTION
**Pull request by megatherium-95**
Before dereferencing mountp argument, check for it's NULLness

Fixes #29704

Signed-off-by: iva kik <megatheriumiva@gmail.com>

### CODE REVIEW

File: tests/subsys/fs/fs_api/src/test_fs.c [hunk 1]
```diff
@@ -73,7 +73,7 @@ static int temp_unlink(struct fs_mount_t *mountp, const char *path)
 }
 
 static int temp_rename(struct fs_mount_t *mountp, const char *from,
-			const char *to)
+		       const char *to)
```
Comment (de-nordic) [inline tests/subsys/fs/fs_api/src/test_fs.c:76]: White space changes are not related to the issue being solved; they should have separate commit.

Comment (megatherium-95) [reply tests/subsys/fs/fs_api/src/test_fs.c]: Fixed

Comment (de-nordic) [reply tests/subsys/fs/fs_api/src/test_fs.c]: If you want to follow through with the whitespace fixes you may another commit on top of this one or create new pull request.

**Review by de-nordic — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
